File: js/task3/S3_PIPE_01.js
// S3_PIPE_01

const pipe = (...fns) => x => fns.reduce((v, f) => f(v), x);

const inc = x => x + 1;
const double = x => x * 2;

// (5 + 1) * 2 = 12
const func = pipe(inc, double);
console.log(func(5));

File: python/task3c/S3_PIPE_01.py
# S3_PIPE_01
from functools import reduce

def pipe(*fns):
    return lambda x: reduce(lambda v, f: f(v), fns, x)

def inc(x): return x + 1
def double(x): return x * 2

# (5 + 1) * 2 = 12
pipeline = pipe(inc, double)
print(pipeline(5))